packages/headless/src/controllers/field-suggestions/facet/headless-field-suggestions.ts

```diff
@@ -1,8 +1,70 @@
 import {SearchEngine} from '../../../app/search-engine/search-engine';
-import {FacetSearch} from '../../core/facets/facet-search/specific/headless-facet-search';
-import {buildFacet} from '../../facets/facet/headless-facet';
-import {FacetOptions} from '../../facets/facet/headless-facet-options';
-import {Subscribable} from '../../controller/headless-controller';
+import {buildFacetSearch} from '../../core/facets/facet-search/specific/headless-facet-search';
+import {
+  buildController,
+  Subscribable,
+} from '../../controller/headless-controller';
+import {loadReducerError} from '../../../utils/errors';
+import {CoreEngine} from '../../../app/engine';
+import {
+  facetSet,
+  configuration,
+  facetSearchSet,
+  search,
+} from '../../../app/reducers';
+import {SearchThunkExtraArguments} from '../../../app/search-thunk-extra-arguments';
+import {
+  FacetSection,
+  ConfigurationSection,
+  FacetSearchSection,
+  SearchSection,
+} from '../../../state/state-sections';
+import {determineFacetId} from '../../core/facets/_common/facet-id-determinor';
+import {updateFacetOptions} from '../../../features/facet-options/facet-options-actions';
+import {executeSearch} from '../../../features/search/search-actions';
+import {logFacetSelect} from '../../../features/facets/facet-set/facet-set-analytics-actions';
+import {defaultFacetOptions} from '../../../features/facets/facet-set/facet-set-slice';
+import {omit} from '../../../utils/utils';
+import {registerFacet} from '../../../features/facets/facet-set/facet-set-actions';
+import {FacetSortCriterion} from '../../../features/facets/facet-set/interfaces/request';
+
+export interface FieldSuggestionsValue {
```

Comment:
Grammatically, I would suggest renaming this to `FieldSuggestion`, but maybe there's a reason I'm not seeing.

Reply:
I prefer not to change it since the method is called `buildFieldSuggestions` and changing that would be breaking.